### ISSUE TITLE
KinesisFirehoseResponse missing Metadata field

### ISSUE THREAD

**Issue by chris-smith-zocdoc** [aws/aws-lambda-dotnet#1345]
### Describe the feature

The `metadata` field of [KinesisFirehoseResponse](https://github.com/aws/aws-lambda-dotnet/blob/master/Libraries/src/Amazon.Lambda.KinesisFirehoseEvents/KinesisFirehoseResponse.cs) is used for the [dynamic partitioning](https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html) feature in Kinesis Firehose 

Reference implementation in golang https://github.com/aws/aws-lambda-go/blob/1ba09afd7523e0d5a8a2e2fb58ddb8ec8b92ebb7/events/firehose.go#L36-L41 added in https://github.com/aws/aws-lambda-go/pull/395

### Use Case

Used to extract partition keys/values for [dynamic partitioning](https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html) feature in Kinesis Firehose 

### Proposed Solution

Add KinesisFirehoseResponseRecordMetadata class with field `partitionKeys` as `Dictionary<string, string>`

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### AWS .NET SDK and/or Package version used

Amazon.Lambda.KinesisFirehoseEvents 2.1.0

### Targeted .NET Platform

.net 6

### Operating System and version

AmazonLinux

**Comment by ashishdhingra** [comment]
@chris-smith-zocdoc Thanks for the PR. Please advise if you have tested this end-to-end in actual scenario.

**Comment by chris-smith-zocdoc** [comment]
@ashishdhingra I have not yet

**Comment by ashishdhingra** [comment]
> @ashishdhingra I have not yet

Somehow looking at the example at https://docs.aws.amazon.com/firehose/latest/dev/dynamic-partitioning.html, the data type of the metadata entry doesn't appear to be `string`, even though Go SDK models it as string.

**Comment by chris-smith-zocdoc** [comment]
@ashishdhingra The JQ expression isn't relevant to this pr, thats an alternative way of providing the partitioning data without the transformation lambda.

I just confirmed this pr works as intended in our test account.


My test lambda returns the partition information like this
```cs
string eventType = ...

output.Metadata = new KinesisFirehoseResponse.FirehoseResponseRecordMetadata
{
        PartitionKeys = new Dictionary<string, string>
        {
            {"event_type", eventType},
        }
};
```

Relevant section of cdk configuration for `CfnDeliveryStream.extendedS3DestinationConfiguration`
```typescript
prefix: "event_type=!{partitionKeyFromLambda:event_type}/year=!{timestamp:yyyy}/month=!{timestamp:MM}/day=!{timestamp:dd}/hour=!{timestamp:HH}/",
dynamicPartitioningConfiguration: {
   enabled: true
},
processingConfiguration: {
    enabled: true,
    processors: [{
      type: 'Lambda',
      parameters: [
        {
          parameterName: 'LambdaArn',
          parameterValue: functionArn,
        }
      ],
    }],
  },
```
Here `event_type` comes from the transformation lambda (`push` in my example), the rest is using variables made available by firehose

Which results in a file under the path
```
s3://<bucket_name>/event_type=push/year=2022/month=10/day=27/hour=00/<filename>.gz
```

**Comment by chris-smith-zocdoc** [comment]
@ashishdhingra I've completed end to end testing, can you please take a look?

**Comment by normj** [comment]
PR has been merged and released in version 2.2.0 of Amazon.Lambda.KinesisFirehoseEvents. Thanks for the PR!
